hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -80,6 +80,9 @@ public void pickAddress() throws Exception {
                 logger.finest("Using public address the same as the bind address: " + publicAddress);
             }
         } catch (Exception e) {
+            if (serverSocketChannel != null) {
+                serverSocketChannel.close();
```

Comment:
Can this create another exception which has to be caught? Do we have a utility method to close a `Closeable` object which may handle this?